sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -205,6 +194,14 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
   protected case class RegularTestCase(
       name: String, inputFile: String, resultFile: String) extends TestCase
 
+  /** An ANSI-related test case. */
+  protected case class AnsiTestCase(
+      name: String, inputFile: String, resultFile: String) extends TestCase with AnsiTest
+
+  /** An analyzer test that shows the analyzed plan string as output. */
+  protected case class AnalyzerTestCase(
```

Comment:
hmm, will we have analyzer-only test? Or you will remove this one when generating all the analyzer golden files?

Reply:
In this PR I made the test runner generate both a `RegularTestCase` and one of these new `AnalyzerTestCase` instances for each of the eligible input `.sql` files.